Feat/system upgrade

alex-struk opened the pull request:
Upgrades node to 22, as well as bootstrap to 5 and other packages

alex-struk committed:
feat: upgraded node to latest version

alex-struk committed:
feat: upgraded node to latest version

kmandryk committed:
feat: add copilotkit

upgrade to react 18
upgrade to typescript 5
upgrade to bootstrap 5
migrate node-sass to sass
change front end build process from grunt to vite

Add copilotkit chat interface

alex-struk committed:
feat: copilotkit backend and front end example

alex-struk committed:
Merge branch 'feat/marketplace-ai' of https://github.com/bcgov/digital_marketplace into feat/marketplace-ai

kmandryk committed:
Create index.html

kmandryk committed:
Merge branch 'feat/marketplace-ai' of https://github.com/bcgov/digital_marketplace into feat/marketplace-ai

alex-struk committed:
feat: bootstrap upgrade wip

alex-struk committed:
feat: bootstrap migration complete

alex-struk committed:
feat: serve front-end from express, ui fixes

alex-struk committed:
feat: system migration cleanup

alex-struk committed:
feat: system migration - fixed markdown html render

alex-struk committed:
feat: system migration - undid type additions - no longer needed

alex-struk committed:
feat: system upgrade - env var fixes

alex-struk committed:
feat: system upgrade - minor cleanup

alex-struk committed:
feat: system upgrade - feedback cleanup pass

alex-struk committed:
feat: system upgrade - aligned lodash to original behavior

alex-struk committed:
feat: system upgrade - minor sonarqube fixes

alex-struk committed:
feat: system upgrade - simplified solution to navlink issue for react upgrade

alex-struk committed:
feat: system upgrade - missed undefined checks

alex-struk committed:
feat: merge

alex-struk committed:
feat: merge upgrade for backend

alex-struk committed:
feat: merge upgrade for front-end

alex-struk committed:
fix: development merge conflict resolution

alex-struk committed:
Merge branch 'development' into feat/system-upgrade

alex-struk committed:
fix: cleanup, package re-org, tsconfig fix

alex-struk committed:
feat: upgraded the remaining packages to latest versions

alex-struk committed:
refactor: remove old eslint config

alex-struk committed:
fix: removed type:module definition

alex-struk committed:
fix: removed no-redeclare eslint rule, undid related change

alex-struk committed:
Merge branch 'development' into feat/system-upgrade

alex-struk committed:
fix: undone edit required for prod build

alex-struk committed:
fix: build fix

alex-struk committed:
fix: build fix - moved moment-timezone to regular dependencies

alex-struk committed:
refactor: removed unused grunt configs

alex-struk committed:
refactor: undone compression removal

IanFonzie committed:
refactor: provide type params for content

IanFonzie committed:
refactor: provide type params for swu opportunities

IanFonzie committed:
refactor: provide type params for twu opportunities

IanFonzie committed:
refactor: provide type params for cwu opportunities

IanFonzie committed:
refactor: provide type params for swu proposals

IanFonzie committed:
refactor: provide type params for swu consensuses

IanFonzie committed:
refactor: provide type params for swu evaluations

IanFonzie committed:
refactor: provide type params for twu proposals

IanFonzie committed:
refactor: provide type params for twu consensuses

IanFonzie committed:
refactor: provide type params for twu proposals

alex-struk committed:
fix: development branch merge fixes

alex-struk committed:
fix: login fix in dev mode

alex-struk committed:
fix: missing css rule

IanFonzie committed:
chore: update pre-commit

IanFonzie committed:
fix: remove as any assertion and display smaller headings

IanFonzie reviewed (commented):
Looks mostly good though I had a few questions and comments.

Questions:
- ~Why did we get rid of type: ‘module’ in package.json? Just found it strange considering we're using a .mjs to define the config.~
- Is there any benefit to maintaining grunt for the back-end?

Comments:
- Styles are off
- ~Test indicator wasn’t displayed on dev, despite env variable being set (haven’t looked into it beyond the variable being set and I didn't experience the issue locally)~
- ~We *need* to be able to enable debugging from process.tsx~
- A lot of deprecation warnings

alex-struk committed:
feat: removed grunt

alex-struk committed:
fix: user page scroll fix

alex-struk committed:
fix: cleared scss warnings, aligned styles

alex-struk commented:
> Looks mostly good though I had a few questions and comments.
> 
> Questions:
> 
> * ~Why did we get rid of type: ‘module’ in package.json? Just found it strange considering we're using a .mjs to define the config.~
> * Is there any benefit to maintaining grunt for the back-end?
> 
> Comments:
> 
> * Styles are off
> * ~Test indicator wasn’t displayed on dev, despite env variable being set (haven’t looked into it beyond the variable being set and I didn't experience the issue locally)~
> * ~We _need_ to be able to enable debugging from process.tsx~
> * A lot of deprecation warnings

- Renamed eslint.config.mts to js ("module" was set in package.json temporarily, but then removed)
- Completely removed grunt
- Adjusted styles for table headings, table text and buttons paddings. Some minor differences remain around `label` components
- SCSS deprecation warnings cleared, except @import statements (these were added to ignore list). Migration instructs to use `@use` statements, but these must always appear at the top, however bootstrap overrides require importing bootstrap after overrides. Additionally, bootstrap uses @import statements internally, so the usage of these statements follows an established pattern. Once bootstrap upgrades their approach, we can restructure our approach to use new approach using @use/@forward statements.

alex-struk committed:
fix: fixed heading sizes on smaller screens

IanFonzie committed:
fix: use bootstrap 4 border-color

IanFonzie committed:
refactor: use gutter utility classes

IanFonzie committed:
fix: adjust props for checkbox

IanFonzie committed:
refactor: adjust checkbox styles to match pre-upgrade styles

IanFonzie committed:
refactor: remove unnecessary font rule

IanFonzie committed:
fix: adjust import order to take advantage of theming

IanFonzie committed:
fix: override default bootstrap formgroup spacing for mb-0 class

IanFonzie committed:
refactor: set height with calc for better input value text alignment

IanFonzie reviewed (approved):
Made my best attempt at matching the styles pre-system upgrade; feel free to take a look or merge.
